### PR TITLE
Add injection for /etc/os-release file

### LINKED ISSUE #79: Populate the /etc/os-release file with cirros information
Hi there. I'm trying to detect cirros containers within my product, and it's a bit tricky since there isn't information in the standard /etc/os-release on cirros. The containers are instead being identified as buildroot. Would it be possible to add the os and release in /etc/os-release so that tools that parse that file could more easily identify cirros containers?

Thanks,
Tim

### PR DESCRIPTION
**Pull request by NucciTheBoss**
Dynamically generate /etc/os-release using a here document in bin/build-release. This way we can grab verison_id, kernel, grub, and buildroot. Also makes it easier to administrate CirrOS images/locate where to report bugs. Example output:

```
$ cat /etc/os-release
PRETTY_NAME="CirrOS d220803"
NAME="CirrOS"
VERSION_ID="d220803"
ID=cirros
HOME_URL="https://github.com/cirros-dev/cirros"
BUG_REPORT_URL="https://github.com/cirros-dev/cirros/issues"
BUILDROOT_VERSION="2019.02.1"
KERNEL_VERSION="5.3.0-26.28~18.04.1"
GRUB_VERSION="2.02-2ubuntu8.14"
```

Closes #79

### CODE REVIEW

**Comment by hrw** [comment]
I like it

File: bin/build-release [hunk 1]
```diff
@@ -179,6 +179,17 @@ ln -snf buildroot-${BR_VER} buildroot
...
+cat << EOF > "src/etc/os-release"
+PRETTY_NAME="CirrOS $VER"
+NAME="CirrOS"
+VERSION_ID="$VER"
+ID=cirros
+HOME_URL="https://github.com/cirros-dev/cirros"
+BUG_REPORT_URL="https://github.com/cirros-dev/cirros/issues"
+BUILDROOT_VERSION="$BR_VER"
+KERNEL_VERSION="$KVER"
+GRUB_VERSION="$GVER"
```
Comment (smoser) [inline bin/build-release:191]: lets drop BUILDROOT_VERSION, KERNEL_VERSION and GRUB_VERSION.

As for my reasoning:
 * They're not standard per a quick google result [os-release(5)](https://manpages.ubuntu.com/manpages/jammy/en/man5/os-release.5.html)
 * I don't have a use-case that they solve or help solve.  What would they be used for?
 * Ubuntu/RHEL/... don't advertise versions of their components in this file.

**Review by smoser — CHANGES_REQUESTED** [review]
Thank you for doing this.

2 small change requests inline and one feature request.

We have previously used /etc/cirros/version to declare the version, and expose that in CIRROS_VERSION when src/lib/cirros/shlib_cirros is sourced.

Could you :
 1. remove /etc/cirros/version. (i would not do this if it were being made in a stable release)
 2. update the 'cirros_version' function in src/lib/cirros/shlib_cirros read /etc/os-release.

Here is a quick updated 'cirros_version' function:

```bash
cirros_version() {
	local orelf="${1:-/etc/os-release}" v="" quote="\""
	_RET="unreleased"
	[ -r "$orelf" ] || return 0;
	while read line ; do
		case "$line" in
			VERSION_ID=*)
				v=${line#*=}
				v=${v%${quote}}
				v=${v#${quote}}
				_RET="$v"
				return 0
				;;
		esac
	done < "$orelf"
	return
}
```
```

**Comment by smoser** [comment]
one other comment. 

cirros's "disk images" are just disk images with an empty filesystem labelled "cirros-rootfs" (from ROOTFS_LABEL in /init).  The kernel and initramfs then copy/"install" themselves to the filesystem (/init lines 57-77).

I just had the thought that it might be nice if these filesystem images *had* a '/etc/os-release' file so that if some thing went looking it would find that.

So in /init lines 57-77  you'd have to amend the "install" to account for an existing /etc/os-release (easiest way to do that would be just to remove it).

I'm fine if you don't do that in this PR (its probalby better if you do not).  But If you wanted to poke around cirros a bit more, that'd be something to do.

Again, thanks!

File: bin/build-release [hunk 2]
```diff
@@ -179,6 +179,17 @@ ln -snf buildroot-${BR_VER} buildroot
 ( cd buildroot && QUILT_PATCHES="$PWD/../patches-buildroot" quilt push -a )
 
 echo "$VER" > "src/etc/cirros/version"
+cat << EOF > "src/etc/os-release"
+PRETTY_NAME="CirrOS $VER"
+NAME="CirrOS"
+VERSION_ID="$VER"
+ID=cirros
+HOME_URL="https://github.com/cirros-dev/cirros"
```
[the remark this answers is not in the document]
Comment (NucciTheBoss) [reply bin/build-release:187]: Sounds good. I originally had it as the original URL, but switched it to the GitHub URL when I saw the redirect as I was not sure if http://cirros-cloud.net/ was still official. Thanks for clearing that up.

File: bin/build-release (hunk 1 above)
Comment (NucciTheBoss) [reply bin/build-release:191]: sgtm. I originally added those variables for extra distro-level metadata (potentially for the curious), but I do agree that there does not seem to be an obvious use-case for this info + it is unstandard information. I will drop this information for now and we can add it back if requested by popular demand.

File: bin/build-release (hunk 2 above)
[the remark this answers is not in the document]
Comment (smoser) [reply bin/build-release:187]: oh. and i just remembered, i have to renew that this month. thanks for making me think about that.

**Comment by hrw** [comment]
@NucciTheBoss can you rebase and finish patch?

Would be great to have proper /etc/os-release in 0.6.0 release.

**Comment by NucciTheBoss** [comment]
@hrw Sorry about my absence, got caught up in a terminal UI and CI project that took up most of my time for the past month. Project wrapped up earlier this week so I can breathe again. 

I will get this out tonight/tomorrow.

**Comment by hrw** [comment]
Can you squash and rebase those changes?

**Comment by NucciTheBoss** [comment]
Well that's not what I wanted

**Comment by NucciTheBoss** [comment]
Going to close out the pull request and submit a new one. Rebase got messed up. I tried to squash the commits by changing the order, but it looks that it included all the other commits as well.

**Pull request closed without merging**